include/linker/devicetree_reserved.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2021, Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Generate memory regions and sections from reserved-memory nodes.
+ */
+
+#include <devicetree.h>
+
+/* Reserved memory node */
+#define _NODE_RESERVED DT_INST(0, reserved_memory)
+
+/* Unquoted region label */
+#define _DT_LABEL_TOKEN(res) DT_STRING_TOKEN(res, label)
+
+/* _start and _end section symbols */
+#define _DT_RESERVED_PREFIX(res) UTIL_CAT(reserved_, _DT_LABEL_TOKEN(res))
+#define _DT_RESERVED_START(res)  UTIL_CAT(_DT_RESERVED_PREFIX(res), _start)
```

Comment:
In your description, and based on this code, you will have section `res0` but linker symbols `reserved_res0_start` and `reserved_res0_end`.

I would like if we could try to establish a pattern where we uses `__<section_name>_start` and `__<section_name>_end`.
This is mostly what we do today and will make it easier to generate linker scripts with known names in the future.

So in this case `res0` --> `__res0_start`, or you could change the name from `res0` --> `reserved0` in DT, and you'd get `__reserved0_start`.

Reply:
I'll change the code to generate `__<section_name>__{start,end}`.